src/plugins/intel_cpu/src/node.cpp

```diff
@@ -842,9 +842,7 @@ void Node::prepareMemory(const DnnlMemoryDescPtr& intDesc, size_t indx) {
     MemoryPtr ptr;
     auto weightCache = context->getWeightsCache();
     if (weightCache != nullptr && memory::format_kind::blocked == intDesc->getDnnlDesc().get_format_kind()) {
-        const auto string_hash =
-            name + "_" + std::to_string(indx) + "_" + DnnlExtensionUtils::computeWeightsStringHash(internalBlob, intDesc);
-        ptr = *weightCache->findOrCreate(string_hash, create);
+        ptr = *weightCache->findOrCreate(DnnlExtensionUtils::computeWeightsStringHash(internalBlob, intDesc, weightCache), create);
```

Comment:
I would prefer to avoid getting back to SimpleDataHash computation mostly due to performance constraints, If the only degradations comes from RNN I would propose the following:
1. Revert SimpleDataHash usage everywhere
2. Reimplement RNN node to use prepareWeightMemory approach (either reuse Node::prepareWeightMemory or implement own version) and get rid of internalBlobs usage
3. FInally remove internalBlobs support from Node class

Reply:
1. SimpleDataHash was removed;
2. internalBlobs was removed from RNN node;
3. internalBlobs was NOT removed from CPU Node due to it affects all prepareMemory functions. I suggest to do this in scope of other task.